dicom_utils/container/record.py

```diff
@@ -0,0 +1,242 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+
+from concurrent.futures import ProcessPoolExecutor, ThreadPoolExecutor
+from dataclasses import dataclass
+from functools import cached_property
+from pathlib import Path
+from typing import Any, Dict, Iterator, List, Optional, Sequence, Set
+
+import pydicom
+from tqdm import tqdm
+
+from ..tags import Tag
+
+# Type checking fails when dataclass attr name matches a type alias.
+# Import types under a different alias
+from ..types import ImageType
+from ..types import PhotometricInterpretation as PI
+from ..types import SimpleImageType as SIT
+from .helpers import SeriesUID, StudyUID
+from .helpers import TransferSyntaxUID as TSUID
+
+
+tags: List[Any] = [
```

Comment:
I converted this to `Final` and added a cast for `pydicom.dcmread`. I chose `Final` over `List[Tag]` since strings or int tuples would also be valid in this list. The `specific_tags` arg in `dcmread` has some annoying type expectations, hence the cast.